src/bosh-director/lib/bosh/director/jobs/ssh.rb

```diff
@@ -21,17 +22,32 @@ def initialize(deployment_id, options = {})
 
       def perform
         target = Target.new(@target_payload)
-
         filter = {}
-        filter[:job] = target.job if target.job
-        filter.merge!(target.id_filter)
+        if target.job && ip_address?(target.job) && instance_count(target.job) == 0
+          ip_address = target.job
+          net_address = ip_to_netaddr(ip_address)
+          ip_error_message = "{:ip=>#{ip_address}}"
+          ipaddress_model = Models::IpAddress.where(address: net_address.to_i).first
+          if ipaddress_model.nil?
+            filter = []
+            filter << Sequel.like(:spec_json, "%\\\"ip\\\":\\\"#{ip_address}\\\"%")
+          else
+            filter[:id] = ipaddress_model.instance_id
+          end
+        else
+          filter[:job] = target.job if target.job
+          filter.merge!(target.id_filter)
+        end
 
         deployment = Models::Deployment[@deployment_id]
-
-        instances = @instance_manager.filter_by(deployment, filter).reject { |i| i.active_vm.nil? }
+        begin
+          instances = @instance_manager.filter_by(deployment, filter).reject { |i| i.active_vm.nil? }
+        rescue InstanceNotFound
```

Comment:
Following above comment: if we isolate the abstraction as described there, wouldn't we be able to have this error be swallowed neatly, so that we don't need to start handling this case.